src/PromiseInterface.php

```diff
@@ -44,7 +48,7 @@ public function then(?callable $onFulfilled = null, ?callable $onRejected = null
      * Since the purpose of `done()` is consumption rather than transformation,
      * `done()` always returns `null`.
      *
-     * @param callable|null $onFulfilled
+     * @param callable(mixed):TResolve|null $onFulfilled
```

Comment:
I'd think this should be `TResolved`, but it's not defined on this method so I don't think that would work? Not sure what this is supposed to do here.